cds_ils/migrator/acquisition/vendors.py

```diff
@@ -29,11 +29,29 @@
 from cds_ils.migrator.utils import bulk_index_records, \
     model_provider_by_rectype
 
+ORIGINAL_WILEY_ID = 17
+DEFAULT_WILEY = 333
+WILEY_DE = 333
+WILEY_UK = 444
+WILEY_US = 555
+WILEY_MAPER = {
+    'EUR': WILEY_DE,
+    'GBP': WILEY_UK,
+    'USD': WILEY_US,
+}
 
-def get_vendor_pid_by_legacy_id(legacy_id):
+
+def get_vendor_pid_by_legacy_id(legacy_id, grand_total):
     """Search for vendor by legacy id."""
+    # Check for Wiley vendor to split it depending on the currency
+    if legacy_id == ORIGINAL_WILEY_ID:
+        if grand_total and grand_total['currency']:
+            legacy_id = WILEY_MAPER.get(grand_total['currency'], DEFAULT_WILEY)
+        else:
+            legacy_id = DEFAULT_WILEY
+
     search = current_ils_acq.vendor_search_cls().filter(
-        "term", legacy_id=legacy_id
+        "term", legacy_ids=legacy_id
```

Comment:
is it tested? I had many problems with the searched after we changed mappings, so I'm asking to be extra sure

Reply:
Yes, here is the proof:
![image](https://user-images.githubusercontent.com/15194802/106203162-420f2700-61bb-11eb-90d4-81b7de809923.png)

Reply:
thanks a lot :)